analysis/02_new_model_input/03.0_download_rainfall_dataset.md

```diff
@@ -0,0 +1,197 @@
+days_to_landfall = 2
+
+# Default = FALSE
+# IMPORTANT: setting to TRUE means that
+# all downloaded GPM files will be deleted and re-downloaded
+delete_folders = False
+```
+
+```python
+#%% Functions used
+def date_range(start_date, end_date):
+    return [
+        str(start_date + dt.timedelta(days=x))
+        for x in range((end_date - start_date).days + 1)
+    ]
+
+
+def listFD(url, user_name, password, ext=""):
+    try:
+        page = requests.get(url, auth=(user_name, password)).text
+        soup = BeautifulSoup(page, "html.parser")
+    except requests.exceptions.RequestException as error:
+        print("Error:\n", error)
+    return [
+        url + "/" + node.get("href")
+        for node in soup.find_all("a")
+        if node.get("href").endswith(ext)
+    ]
+
+
+def download_gpm_http(start_date, end_date, download_path, type_imerg):
+    # Username and password for logging in
+    # Can create own account on NASA site
+    # user_name = "pauline.ndirangu@un.org"#GPM_USERNAME
+    base_url = ""
+
+    # Connection to the site, if pasting in chrome:
+    # https://arthurhouhttps.pps.eosdis.nasa.gov/
+    # Directory to where the data is saved
+    if type_imerg == "final":
+        base_url = baseurl
+
+    date_list = date_range(start_date, end_date)
+    file_list = []
+
+    for date in date_list:
+        print(date)
+        d, m, y = reversed(date.split("-"))
+        day_path = os.path.join(download_path, y + m + d)
+
+        # Make a folder for each day, to save GPM data
+
+        os.makedirs(day_path, exist_ok=True)
+        if type_imerg == "final":
+            data_dir_final = f"/{y}/{m}/{d}/gis"
+            url = base_url + data_dir_final
+            tiff_files = listFD(
+                url, user_name=user_name, password=user_name, ext="tif"
+            )
+
+            for tifffile in tiff_files:
+                file_name = tifffile.split("/")[-1]
+
+                file_path = os.path.join(day_path, file_name)
+                file_list.append(file_path)
+                try:
+                    r = requests.get(tifffile, auth=(user_name, user_name))
+                    open(file_path, "wb").write(r.content)
+                except requests.exceptions.RequestException as error:
+                    print("Error:\n", error)
+
+    return file_list
+```
+
+```python
+# To make sure the dates can be converted to date type
+for i in range(len(typhoon_metadata)):
+    typhoon_metadata["startdate"][i] = typhoon_metadata["startdate"][
+        i
+    ].replace("/", "-")
+    typhoon_metadata["enddate"][i] = typhoon_metadata["enddate"][i].replace(
+        "/", "-"
+    )
+    typhoon_metadata["landfalldate"][i] = typhoon_metadata["landfalldate"][
+        i
+    ].replace("/", "-")
+```
+
+```python
+# Typhoons for which to download rainfall data
+typhoons = list(typhoon_metadata.typhoon.values)
+
+# if there is already rainfall data in the project folder  output_hhr_processed
```

Comment:
I feel like you can just combine everything from here below into a single for loop, it's confusing that it's split up into three. 

Also I think it's somewhat dangerous that it automatically doesn't re-download a directory, you may end up with a directory with only half the dates or something like that. Instead you can implement a file-wise check, although I'm not sure it's needed.